Description: [droidcamsrc] send flush stop when transition to playing
 vfsrc pad is set to be flushing when droidcamsrc transitions from
 playing to paused. Revert this when transitions back to playing, or
 the mode activation will fail as it tries to negotiate & set pad's caps.
Author: Ratchanan Srirattanamet <ratchannan@ubports.com>
Forwarded: https://github.com/sailfishos/gst-droid/pull/52https://github.com/sailfishos/gst-droid/pull/52
Last-Update: 2020-06-10
---
This patch header follows DEP-3: http://dep.debian.net/deps/dep3/
--- a/gst/droidcamsrc/gstdroidcamsrc.c
+++ b/gst/droidcamsrc/gstdroidcamsrc.c
@@ -700,6 +700,15 @@
       /* set initial photography parameters */
       gst_droidcamsrc_photography_apply (src, SET_ONLY);
 
+      /* send flush stop if we start it in the previous transition */
+      if (GST_PAD_IS_FLUSHING (src->vfsrc->pad)) {
+        if (!gst_pad_push_event (src->vfsrc->pad,
+                gst_event_new_flush_stop ( /* reset_time */ TRUE))) {
+          ret = GST_STATE_CHANGE_FAILURE;
+          break;
+        }
+      }
+
       /* activate mode */
       if (!gst_droidcamsrc_select_and_activate_mode (src)) {
         ret = GST_STATE_CHANGE_FAILURE;
